harness/tests/determined/common/experimental/test_trial.py

```diff
@@ -0,0 +1,74 @@
+import datetime
+from typing import Callable
+
+import pytest
+import responses
+
+from determined.common import api
+from determined.common.experimental import trial
+from tests.fixtures import api_responses
+
+_MASTER = "http://localhost:8080"
+
+
+@pytest.fixture
+def standard_session() -> api.Session:
+    return api.Session(master=_MASTER, user=None, auth=None, cert=None)
+
+
+@pytest.fixture
+def make_trialref(standard_session: api.Session) -> Callable[[int], trial.Trial]:
+    def _make_trialref(trial_id: int) -> trial.Trial:
+        return trial.Trial(trial_id, standard_session)
+
+    return _make_trialref
+
+
+@responses.activate
+def test_trial_logs_converts_epoch_time(make_trialref: Callable[[int], trial.Trial]) -> None:
+    trialref = make_trialref(1)
+    trial_logs = api_responses.sample_get_trial_logs()
+    trial_logs_url = f"{_MASTER}/api/v1/trials/{trialref.id}/logs"
+
+    before_ts = 1689285229
+    after_ts = 1689280229
+
+    timestamp_before = datetime.datetime.fromtimestamp(before_ts).isoformat("T") + "Z"
+    timestamp_after = datetime.datetime.fromtimestamp(after_ts).isoformat("T") + "Z"
+    params = {
+        "timestampBefore": timestamp_before,
+        "timestampAfter": timestamp_after,
+    }
+    responses.get(
+        trial_logs_url,
+        stream=True,
+        body=trial_logs,
+        match=[responses.matchers.query_param_matcher(params, strict_match=False)],
+    )
+
+    resp = trialref.logs(timestamp_before=before_ts, timestamp_after=after_ts)
+
+    assert len(list(resp)) > 1
```

Comment:
I think there are a couple extraneous assertions, here, when comparing with the test's name. I think the only assertion(s) that need to be made to satisfy the test as named are
```python
call.request.params["timestampBefore"] = timestamp_before
call.request.params["timestampAfter"] = timestamp_after
```

Since "conversion" is all we're testing, I wonder if the better place to insert the check is into 'determined.common.api.logs.trial_logs`. This question is kind of making me wonder if heretofore we should have been putting more attention into mocking bindings and less into faking the REST API.

What do you think?

Reply:
discussed this offline. i think this is a good idea, but i'll defer attempting binding mocks to another feature.